---
title: "06-transcriptomics"
output: html_document
---
# Background
These samples were sequenced by NovoGene in paired-end mode, 150 bp
reads. We used
[fastqc](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
to assess the overall quality for the [FASTQ](multiqc_report.html)
files. The link will open a summary of the fastqc results, which
indicate that the raw data are of high quality.

We aligned reads to the Botryllus_tozio_genes.fasta transcript file
provided by the Gardell lab, using the salmon aligner
[@patro2015salmon], and then summarized to the gene level using the
Bioconductor tximport package [@tximport].

The FASTA transcripts file appears to have been generated by the
Trinity aligner, which starts by identifying transcripts and then
collapsing the transcripts to genes. The resulting identifier formats
are something like g1.t1, which in this case indicates that the given
transcript is the 'first' (t1) transcript for the 'first' (g1) gene.

The transcripts were annotated using both eggNOG and OmicsBox. the
eggNOG annotations were more complete, so we used those to map the
genes to gene symbols and GO terms.