---
title: "Extracting Transcript Sequences"
output: html_document
date: "2025-03-26"
---
# Extract Transcript Sequences and make a new FASTA file

```{r, engine= 'bash'}
#!/bin/bash

# Input FASTA file containing the full transcriptome
INPUT_FASTA="/home/shared/8TB_HDD_02/cvaldi/Botryllus-Nickel-Tox/data/Botryllus_tozio_genes.fasta"

# Gene ID list file
GENE_IDS="/home/shared/8TB_HDD_02/cvaldi/Botryllus-Nickel-Tox/data/gene_ids.txt"  # List of gene IDs you want to extract

# Output FASTA file for the extracted sequences
OUTPUT_FASTA="/home/shared/8TB_HDD_02/cvaldi/Botryllus-Nickel-Tox/data/noname_sequences.fasta"  # Output file

# Extract ids and gene ids into a tsv file:
perl -lne '@f = /^>(\S+)\s+(\S+)/ and print join "\t", @f;' $INPUT_FASTA > ids_gene_ids.tsv

# Select ids that correspond to the desired gene ids (from gene_ids.txt):
grep -f $GENE_IDS ids_gene_ids.tsv | cut -f1 > ids.selected.txt

# Use seqkit to extract the sequences that correspond to the gene ids:
seqkit subseq $INPUT_FASTA ids.selected.txt > $OUTPUT_FASTA

echo "Extraction complete. Sequences saved in $OUTPUT_FASTA."

# Optionally, view the output file:
cat $OUTPUT_FASTA

```

```{r, engine= 'bash'}
#!/bin/bash

# Input FASTA file containing the full transcriptome
INPUT_FASTA="../data/Botryllus_tozio_genes.fasta"  # Update to your input FASTA file path

# Gene ID list file
GENE_IDS="../data/gene_ids.txt"  # List of gene IDs you want to extract (one per line)



# Output FASTA file for the extracted sequences
OUTPUT_FASTA="../data/noname_sequences.fasta"  # Output file

# Clear the output file if it exists
> $OUTPUT_FASTA

# Loop through each gene ID in the list and extract its sequence
while IFS= read -r gene; do
    # Use awk to extract sequences for each gene_id
    awk -v gene="$gene" '/^>'gene'[^0-9]/{flag=1; print; next} /^>/{flag=0} flag' $INPUT_FASTA >> $OUTPUT_FASTA
    echo "Extracted sequence for: $gene"
done < "$GENE_IDS"

echo "Extraction complete. Sequences saved in $OUTPUT_FASTA."

awk '/^>g10908/{flag=1; print; next} /^>/{flag=0} flag' $INPUT_FASTA


```





# Manually extracting each sequence
We will now pull the transcript sequences of the genes that are unannotated. Below you can use what is spit out in the console copy and paste them into blastx to find associated proteins. You can then update the file noname.csv by hand.
```{r, engine='bash'}

awk '/^>g1294[^0-9]/{flag=1; print; next} /^>/{flag=0} flag' ../data/Botryllus_tozio_genes.fasta

```


```{r, engine='bash'}

awk '/^>g1294/{flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta

```

```{r, engine='bash'}

awk '/^>g6611/{flag=1; print; next} /^>/{flag=0} flag' ../data/Botryllus_tozio_genes.fasta

```

```{r, engine='bash'}

awk '/^>g1196[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta

```

```{r, engine='bash'}
awk '/^>g12050[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g12111[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta

```

```{r, engine= 'bash'}
awk '/^>g12775[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g1294[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g12982[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g13014[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g14195[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g14703[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g14965[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g15728[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```
```{r, engine= 'bash'}
awk '/^>g15986[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```
```{r, engine= 'bash'}
awk '/^>g16191[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g16471[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```
```{r, engine= 'bash'}
awk '/^>g1703[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g1744[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g2472[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g2473[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g4208[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```
```{r, engine= 'bash'}
awk '/^>g5661[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g6611[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```
```{r, engine= 'bash'}
awk '/^>g7259[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g7613[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g8170[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g9341[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```

```{r, engine= 'bash'}
awk '/^>g9597[^0-9]/ {flag=1; print; next} /^>/{flag=0} flag{printf "%s", $0} END {print ""}' ../data/Botryllus_tozio_genes.fasta
```